cedar-policy/src/proto/validator.rs

```diff
@@ -0,0 +1,436 @@
+    }
+}
+
+impl From<&models::ValidatorActionId> for cedar_policy_validator::ValidatorActionId {
+    // PANIC SAFETY: experimental feature
+    #[allow(clippy::expect_used)]
+    fn from(v: &models::ValidatorActionId) -> Self {
+        let extensions_none = extensions::Extensions::none();
+        let eval = evaluator::RestrictedEvaluator::new(extensions_none);
+        Self::new(
+            ast::EntityUID::from(v.name.as_ref().expect("name field should exist")),
+            v.applies_to
+                .as_ref()
+                .expect("applies_to field should exist")
+                .principal_apply_spec
+                .iter()
+                .map(ast::EntityType::from),
+            v.applies_to
+                .as_ref()
+                .expect("applies_to field should exist")
+                .resource_apply_spec
+                .iter()
+                .map(ast::EntityType::from),
+            v.descendants.iter().map(ast::EntityUID::from),
+            types::Type::from(v.context.as_ref().expect("context field should exist")),
+            types::Attributes::from(
+                v.attribute_types
+                    .as_ref()
+                    .expect("attribute_types field should exist"),
+            ),
+            v.attributes
+                .iter()
+                .map(|(k, v)| {
+                    let pval = eval
+                        .partial_interpret(
+                            ast::BorrowedRestrictedExpr::new(&ast::Expr::from(v))
+                                .expect("RestrictedExpr"),
+                        )
+                        .expect("interpret on RestrictedExpr");
+                    (k.into(), pval.into())
+                })
+                .collect(),
+        )
+    }
+}
+
+impl From<&cedar_policy_validator::ValidatorEntityType> for models::ValidatorEntityType {
+    fn from(v: &cedar_policy_validator::ValidatorEntityType) -> Self {
+        let name = Some(models::EntityType::from(v.name()));
+        let descendants = v.descendants.iter().map(models::EntityType::from).collect();
+        let attributes = Some(models::Attributes::from(v.attributes()));
+        let open_attributes = models::OpenTag::from(&v.open_attributes()).into();
+        let tags = v.tag_type().map(|tags| models::Tag {
+            optional_type: Some(models::Type::from(tags)),
+        });
+        match &v.kind {
+            cedar_policy_validator::ValidatorEntityTypeKind::Standard(_) => Self {
+                name,
+                descendants,
+                attributes,
+                open_attributes,
+                tags,
+                enum_choices: vec![],
+            },
+            cedar_policy_validator::ValidatorEntityTypeKind::Enum(enum_choices) => Self {
+                name,
+                descendants,
+                attributes,
+                open_attributes,
+                tags,
+                enum_choices: enum_choices.into_iter().map(ToString::to_string).collect(),
```

Comment:
Can't you use `new_enum` and `new_standard` here for some reason?

Reply:
Those are methods for constructing `cedar_policy_validator::ValidatorEntityType`.  This code is constructing `models::ValidatorEntityType`.